src/sciline/pipeline.py

```diff
@@ -194,3 +199,93 @@ def bind_and_call(
     def _repr_html_(self) -> str:
         nodes = ((key, data) for key, data in self._graph.nodes.items())
         return pipeline_html_repr(nodes)
+
+
+def get_mapped_node_names(
+    graph: Pipeline, key: type, index_names: Sequence[Hashable] | None = None
+) -> pandas.Series:
+    """
+    Given a graph with key depending on mapped nodes, return a series corresponding
+    mapped keys.
+
+    This is meant to be used in combination with :py:func:`Pipeline.map`.
+    If the key depends on multiple indices, the series will be a multi-index series.
+
+    Note that Pandas is not a dependency of Sciline and must be installed separately.
+
+    Parameters
+    ----------
+    graph:
+        The pipeline to get the mapped key names from.
+    key:
+        The key to get the mapped key names for. This key must depend on mapped nodes.
+    index_names:
+        Specifies the names of the indices of the mapped node. If not given this is
+        inferred from the graph, but the argument may be required to disambiguate
+        multiple mapped nodes with the same name.
+
+    Returns
+    -------
+    :
+        The series of mapped key names.
+    """
+    import pandas as pd
+    from cyclebane.graph import IndexValues, MappedNode, NodeName
+
+    candidates = [
+        node
+        for node in graph._cbgraph.graph.nodes
+        if isinstance(node, MappedNode) and node.name == key
+    ]
+    if index_names is not None:
+        candidates = [
+            node for node in candidates if set(node.indices) == set(index_names)
+        ]
+    if len(candidates) == 0:
+        raise ValueError(f"'{key}' is not a mapped node.")
```

Comment:
Should this check be before filtering by index names? It seems that `key` refers to a mapped node but that is filtered out by the index names.